specification/evaluation-context/evaluation-context.md

```diff
@@ -1,5 +1,41 @@
 # Evaluation Context
 
+**Status**: [Experimental](../README.md#document-statuses)
+
 ## Overview
 
-// TODO: this is a stub.
+The `evaluation context` provides ambient information for the purposes of flag evaluation. Contextual data may be used as the basis for targeting, including rule-based evaluation, overrides for specific subjects, or fractional flag evaluation.
+
+### Fields
+
+##### Requirement 3.1
+
+> The `evaluation context` structure **MUST** define a required `targeting key` field of type string, identifying the subject of the flag evaluation.
+
+The targeting key uniquely identifies the subject (end-user, or client service) of a flag evaluation. Providers may use this as a unique identifier for fractional-flags, rules, or overrides targeting specific users.
+
+##### Requirement 3.2
+
+> The evaluation context **MUST** define the following optional fields: `email` (string), `first name` (string), `last name`(string), `name`(string), `client ip`(string), `tz`(string), `locale`(string), `timestamp`(date), `environment`(string), `application`(string), and `version`(string).
+
+See the table below for details.
+
+| Name        | Type     | Purpose                                   | Example(s)                                                |
+| ----------- | -------- | ----------------------------------------- | --------------------------------------------------------- |
+| email       | string   | Subject RFC 5322 conformant email         | "first.last@example.com"                                  |
+| first name  | string   | Subject given name                        | "Jane"                                                    |
+| last name   | string   | Subject last name                         | "Smith"                                                   |
+| name        | string   | Subject full name and titles and suffixes | "Ms. Jane Smith"                                          |
+| client ip   | string   | Client IP address (IPv4 or IPv6)          | "216.7.132.218","2001:0db8:85a3:0000:0000:8a2e:0370:7334" |
+| tz          | string   | Subject "zoneinfo" timezone               | "Brazil/West","America/New_York"                          |
+| locale      | string   | Subject RFC 5646 locale                   | "pt-BR","en-US"                                           |
+| timestamp   | datetime | The time of event/request                 | "2022-05-10T01:53:29.823Z"                                |
```

Comment:
Unleashed has a `currentTime` field. I guess there could be a use case for this. I've added the stipulation that it should be in client local time, as I think that could be useful.

Does that make sense? Is it too prescriptive? Do we even really need this field?

Reply:
I could go either way but it doesn't seem necessary to me. It looks like Unleash uses it for time based releases but it defaults to the current time if nothing is set.

Reply:
This seems superfluous to me, and also opens a rabbit hole of time zones, time formats etc etc

Often epoch time is more useful when driving flag rules as it means simple < > against ints rather than manipulating datetime formats. But that can just be set as a regular int. So I guess my vote would be to not have it. It's also implicit in terms of when the request was sent?

Reply:
Couldn't the provide just send the system time? Would you ever want to send a time that isn't the current time? Even if you did, excluding it from the semantics here wouldn't prevent you from doing that. I would recommend we omit it now and reevaluate if someone opens an issue in the future.

Reply:
>Couldn't the provide just send the system time? Would you ever want to send a time that isn't the current time?

I REALLY don't think it's a good idea for providers to add data to the context themselves, especially in the "official" fields. In fact, I was going to add that as a point to the spec. I think this is a PII concern.

Reply:
I've removed the `timestamp` field.